Bottom: 24c795e5bad7d7c53b2ef21274f37d815ce26584
Top:    261d6f64bb34ac1c1d82308b5bdd4febd51c767f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-09 12:29:07 +0100

Revert "We can just delete kicked out holes"

This reverts commit 6400e974907264b9a0524a0dbdd7dfb9eb623f8b.


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 4a761d5..e7642c9 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1423,8 +1423,7 @@ InsertHoles (jmp_buf * e, POLYAREA * dest, PLINE ** src)
 	}
       else
 	{
-	  /* Need to check if this new hole means we need to kick out any old
-	   * ones for deletion */
+	  /* Need to check if this new hole means we need to kick out any old ones for reprocessing */
 	  while (1)
 	    {
 	      struct find_inside_info info;
@@ -1458,7 +1457,10 @@ InsertHoles (jmp_buf * e, POLYAREA * dest, PLINE ** src)
 
 	      /* Remove hole from the contour */
 	      remove_contour (pa_info->pa, prev, info.result, TRUE);
-	      poly_DelContour (&info.result);
+
+	      /* Add hole as the next on the list to be processed in this very function */
+	      info.result->next = *src;
+	      *src = info.result;
 	    }
 	  /* End check for kicked out holes */
